transaction-status/src/lib.rs

```diff
@@ -378,16 +378,14 @@ impl UiTransactionStatusMeta {
             post_token_balances: meta
                 .post_token_balances
                 .map(|balance| balance.into_iter().map(Into::into).collect()),
-            rewards: meta.rewards,
+            rewards: if show_rewards { meta.rewards } else { None },
             loaded_addresses: None,
             return_data: meta.return_data,
             compute_units_consumed: meta.compute_units_consumed,
         }
     }
-}
 
-impl From<TransactionStatusMeta> for UiTransactionStatusMeta {
-    fn from(meta: TransactionStatusMeta) -> Self {
+    fn from(meta: TransactionStatusMeta, show_rewards: bool) -> Self {
```

Comment:
I think I prefer to keep the `From` trait impl and instead suggest mutating the result of `UiTransactionStatusMeta::from(meta)` to set the `rewards` field to `None` if `showRewards` is false. Thoughts?

Reply:
Yes, sure. I considered that, but opted for this to avoid having to assign the mutable variable. My preference for this approach is only very slight though, so I'll rework that bit.

Reply:
Done